### ISSUE TITLE
Infinite loop leads to segfault

### ISSUE THREAD

**Issue by jlaine** [qxmpp-project/qxmpp#186]
While running some tests on a shaky connection I ran into what looks like an infinite loop, leading to a crash.

```
...
Thu Jan 17 20:30:31 2019 INFO Connecting to xmpp.jabber.at:443
QAbstractSocket::connectToHost() called when already looking up or connecting/connected to "xmpp.jabber.at"
Thu Jan 17 20:30:31 2019 WARNING Socket error: Trying to connect while connection is in progress
Thu Jan 17 20:30:31 2019 INFO Connecting to xmpp.jabber.at:443
QAbstractSocket::connectToHost() called when already looking up or connecting/connected to "xmpp.jabber.at"
Thu Jan 17 20:30:31 2019 WARNING Socket error: Trying to connect while connection is in progress
Thu Jan 17 20:30:31 2019 INFO Connecting to xmpp.jabber.at:443
QAbstractSocket::connectToHost() called when already looking up or connecting/connected to "xmpp.jabber.at"
Thu Jan 17 20:30:31 2019 WARNING Socket error: Trying to connect while connection is in progress
Thu Jan 17 20:30:31 2019 INFO Connecting to xmpp.jabber.at:443
QAbstractSocket::connectToHost() called when already looking up or connecting/connected to "xmpp.jabber.at"
Thu Jan 17 20:30:31 2019 WARNING Socket error: Trying to connect while connection is in progress
Thu Jan 17 20:30:31 2019 INFO Connecting to xmpp.jabber.at:443
Segmentation fault
```

Might be related to https://github.com/qxmpp-project/qxmpp/pull/105

**Comment by jankusanagi** [comment]
I've seen this too, with QXMPP 1.0.0 (unsure if 0.9.3 had this same issue)

**Comment by lnjX** [comment]
I recently had the same issue, in my case the DNS of the XMPP server was misconfigured / the first SRV record didn't work. I had one SRV record for port 443 (where sslh listens) and one for the normal port 5269. sslh was not running on the server, so the first connection attempt failed, then the infinite loop started.

**Comment by jlaine** [comment]
This seems like a really severe issue which needs to be fixed ASAP

**Comment by lnjX** [comment]
Closed by #216.
